recipe-server/client/control/tests/state/actions/test_reducers.js

```diff
@@ -0,0 +1,28 @@
+import { fromJS } from 'immutable';
+
+import {
+  ACTION_RECEIVE,
+} from 'control/state/action-types';
+import actionsReducer from 'control/state/actions/reducers';
+
+import {
+  INITIAL_STATE,
+  ACTION,
+} from '.';
+
+
+describe('Actions reducer', () => {
+  it('should return initial state by default', () => {
+    expect(actionsReducer(undefined, {})).toEqual(INITIAL_STATE);
+  });
+
+  it('should handle ACTION_RECEIVE', () => {
+    expect(actionsReducer(undefined, {
+      type: ACTION_RECEIVE,
+      action: ACTION,
+    })).toEqual({
+      ...INITIAL_STATE,
+      items: INITIAL_STATE.items.set(ACTION.id, fromJS(ACTION)),
```

Comment:
This feels a bit like writing the same code twice as a test... but it tests the matching, and behavior over time, and the only other thing I can think of would be doing a `fromJS` over replacement initial state, which doesn't seem great.

I think this is just what feelings feel like. I'm with this.